lib/fluent/plugin/in_tail.rb

```diff
@@ -494,9 +500,9 @@ def on_notify
             begin
               while true
                 if @buffer.empty?
-                  @io.read_nonblock(2048, @buffer)
+                  @io.readpartial(2048, @buffer)
```

Comment:
Ah, sure, in this case input is always a file and it reaches EOF; thank you for explanation.

> When on_notify is called by TimerWatcher, file may have 0 bytes.

When readpartial for 0 byte file or already reached EOF, it raises EOFError.
So it won't block.